device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1028,6 +1031,35 @@ public void unregisterMultiplexedDeviceClient(List<ClientConfiguration> configs,
                 this.multiplexedDeviceConnectionStates.remove(newlyUnregisteredConfig.getDeviceId());
             }
         }
+
+        // when a client is unregistered, remove all "waiting" and "in progress" messages that it had queued
+        for (IotHubTransportPacket waitingPacket : this.waitingPacketsQueue)
+        {
+            String deviceIdForMessage = waitingPacket.getDeviceId();
+            for (ClientConfiguration unregisteredConfig : configs)
+            {
+                if (unregisteredConfig.getDeviceId().equals(deviceIdForMessage))
+                {
+                    this.waitingPacketsQueue.remove(waitingPacket);
+                    this.addToCallbackQueue(waitingPacket);
```

Comment:
Based on the comment, I don't understand why it is being removed from this queue.

Reply:
The unsent message here is being removed from the waitingPacketsQueue and added to the callbacks queue so that the user is notified via callback that the message was cancelled

Reply:
This is analogous to how we cancel all queued or in progress messages for a non-multiplexed client if the user closes that client